lib/portlayer/storage/image_cache.go

```diff
@@ -171,6 +169,12 @@ func (c *NameLookupCache) CreateImageStore(op trace.Operation, storeName string)
 	// Create the root image
 	scratch, err := c.DataStore.WriteImage(op, &Image{Store: store}, Scratch.ID, nil, "", nil)
 	if err != nil {
+		// if we failed here, remove the image store
+		op.Infof("Removing failed image store %s", storeName)
+		if e := c.DataStore.DeleteImageStore(op, storeName); e != nil {
+			op.Errorf("image store cleanup failed: %s", e.Error())
```

Comment:
The only scenario in which this will fail legitimately is because it has images in there.  And we dont enter this path if there are images in that directory.

The other path is simply calls to the datastore failing.  THere's really nothing we can do about that since they aren't tasks or bound by some `state` being true.  VSAN is the most tempermental about this stuff and it's pretty OK nuking empty dirs but we'll see if we need a retry around all of the datastore calls soon enough.